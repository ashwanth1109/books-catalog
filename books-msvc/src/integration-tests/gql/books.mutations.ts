export const ADD_BOOK = `
    mutation AddBookMutation($book: BookInput!) {
      addBook(book: $book) {
        title,
        description,
        year
      }
    }
`;
