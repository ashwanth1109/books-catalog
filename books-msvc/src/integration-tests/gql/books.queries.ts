import { gql } from 'apollo-server-core';

export const GET_BOOKS = gql`
  query GetAllBooks {
    books {
      _id
      title
      description
      year
    }
  }
`;

export const GET_BOOK = gql`
  query GetBookById($bookId: ObjectId!) {
    book(bookId: $bookId) {
      _id
      title
      description
      year
    }
  }
`;
