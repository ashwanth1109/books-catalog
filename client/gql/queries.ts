import { gql } from "@apollo/client";

export const GET_BOOK_BY_ID = gql`
  query GetBookById($bookId: ObjectId!) {
    book(bookId: $bookId) {
      _id
      title
      description
      year
    }
  }
`;

export const GET_ALL_BOOKS = gql`
  query GetAllBooks {
    books {
      _id
      title
    }
  }
`;
