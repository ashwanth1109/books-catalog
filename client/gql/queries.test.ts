import { GET_ALL_BOOKS, GET_BOOK_BY_ID } from "gql/queries";

describe("gql.queries unit test", () => {
  describe("GET_BOOK_BY_ID", () => {
    it("should assert that query string is correct", () => {
      expect(GET_BOOK_BY_ID).toMatchSnapshot();
    });
  });

  describe("GET_ALL_BOOKS", () => {
    it("should assert that query string is correct", () => {
      expect(GET_ALL_BOOKS).toMatchSnapshot();
    });
  });
});
