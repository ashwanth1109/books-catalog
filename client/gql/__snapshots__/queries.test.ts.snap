// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gql.queries unit test GET_ALL_BOOKS should assert that query string is correct 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "GetAllBooks",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "books",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "_id",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "title",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 67,
    "start": 0,
  },
}
`;

exports[`gql.queries unit test GET_BOOK_BY_ID should assert that query string is correct 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "GetBookById",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "bookId",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "bookId",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "book",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "_id",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "title",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "description",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "year",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "ObjectId",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "bookId",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 132,
    "start": 0,
  },
}
`;
